packages/domains-table/src/use-basic-domain-data-query.ts

```diff
@@ -0,0 +1,42 @@
+import { UseQueryOptions, useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { BasicDomainData } from './types';
+
+interface QueryFnData {
+	domains: BasicDomainData[];
+}
+
+const EMPTY_STATE = Object.freeze( {} );
+
+export function useBasicDomainDataQuery(
+	options: UseQueryOptions< QueryFnData, unknown, BasicDomainData[] > = {}
+) {
+	const { capabilities, sites } = useSelector( ( state: any ) => ( {
```

Comment:
Is there any way to avoid using `any` here? Maybe [`AppState`](https://github.com/Automattic/wp-calypso/blob/add/domains-table-skeleton/client/types.ts#L109) which is effectively `any` right now but will resolve to a proper type whenever we feel it's appropriate 😄 

We could also shape `state` so that it returns the value of [`getCurrentUser`](https://github.com/Automattic/wp-calypso/blob/add/domains-table-skeleton/client/state/current-user/selectors.js#L27) and [`getSites`](https://github.com/Automattic/wp-calypso/blob/add/domains-table-skeleton/client/state/selectors/get-sites.js).

But the more I think about this hook, the less I'm convinced we should care about this selector's data. We're creating a semi-agnostic package that depends on Calypso data, so why not just move it under Calypso? If we want to make it truly agnostic, we should accept `sitesUserCanManage` as a prop (and probably rename that prop to be just sites). Or just accept a custom `select` function to return different data from the hook.

>I think it's fine for now because we should eventually filter out the unwanted domains on the server side https://github.com/Automattic/dotcom-forge/issues/3434

So maybe we just not filter at all, and assume it will come correctly from the back-end? This feels like duplicate work for now and something that will be discarded anyway 🤔

Reply:
Yeah `AppState` would be better. Unfortunately it's defined in `/client`, not a package.

> So maybe we just not filter at all, and assume it will come correctly from the back-end? This feels like duplicate work for now and something that will be discarded anyway 🤔

Yep this is exactly how I think about it. It's just code that I hope to throw away eventually, so it's not something I'm worried about getting perfect.

Reply:
That's true, and this is also behind a FF. This is good to go then!